### ISSUE TITLE
FauxGrainTracker and GrainTracker input parameters should be moved to GrainTrackerInterface

### ISSUE THREAD

**Issue by dschwen** [idaholab/moose#6115]
Also some comments need to be fixed. 

FauxGrainTracker looks overly complicated to me. I may not be grasping this fully yet, but what I need is a simple call to get the _grain number_ for a given _order parameter_ (preferably given by variable number) on the _current_ element (yeah, the UserObject may not know what that is, unless it is stored in FEProblem). This will be a heavily used function. The interface where a bool flag (for variable coloring or not) is passed in does not seem very elegant to me.

Ping @permcody

**Comment by permcody** [comment]
Correct. I have a branch with a new interface. The current one is terrible.
It's the product of several add-ons. I plan to push it soon.

**Comment by dschwen** [comment]
FauxGrainTracker is also potentially broken in parallel. I'm getting

```
Assertion `_subproblem.mesh().isSemiLocal(const_cast<Node *>(&node))' failed
Node is not Semilocal
at /Users/schwd/Programs/INL/moose/framework/src/base/MooseVariable.C, line 1593
```

I will have to recompile libmesh with `--enable-tracefiles` to investigate this further, but [this line](https://github.com/idaholab/moose/blob/devel/modules/phase_field/src/postprocessors/FauxGrainTracker.C#L106) might be the cause.

**Comment by dschwen** [comment]
Actually it seems to be [this one](https://github.com/idaholab/moose/blob/devel/modules/phase_field/src/postprocessors/FauxGrainTracker.C#L126) according to the tracefile.

**Comment by permcody** [comment]
You are absolutely right. The element loop needs to go over local active elements, not active elements.

**Comment by permcody** [comment]
Closed by #5740
